codegen: Don't strip space/nl from nodes

camscale opened the pull request:
Do not strip spaces or newlines from nodes in the AST when generating Go
code. If there is a space or newline in the AST, then it should be there
in the generated code. The codegen should just make sure that the
emitted code will compile and do whatever escaping is necessary to
ensure that.

Fixes: #59

camscale commented:
This is [WIP] while I figure out how to unbreak the `go:generate` breakage I introduced previously.

andrewemeryanz commented:
The change looks correct (at cursory glance). Can you add some test examples that demonstrate the change?

camscale commented:
@andrewemeryanz Are you asking that I add test cases for the codegen `grammar.go` (of which there are currently none) or add some sort of example output to the `/examples` directory (which currently contain only `.wbnf` files), or just provide some concrete output in this PR or the linked issue?

Also, when I sort out the `go:generate` stuff, this PR will have the new `wbnf/wbnfgrammar.go` file generated with the new string formatting introduced in this PR, so that will be the original example I guess.

andrewemeryanz commented:
@camh-anz Yes, `grammar.go`. Testing `MakeGrammarString` with a grammar containing a regex that previously caused a panic would test this change. Or are you saying that this is unnecessary because the `wbnf/wbnfgrammar.go` would be generated containing such a regex?

Commit:
codegen: Don't strip space/nl from nodes

Do not strip spaces or newlines from nodes in the AST when generating Go
code. If there is a space or newline in the AST, then it should be there
in the generated code. The codegen should just make sure that the
emitted code will compile and do whatever escaping is necessary to
ensure that.

Issues: #59

camscale commented:
@andrewemeryanz I've reduced the scope of the change to address just the bug and added some test cases for the parts where the bug was impacting the sysl grammar. Given there were no tests there at all, I did not want to write a comprehensive set of tests for stuff I don't know very well.

PTAL
